tests/featured-speaker.spec.ts

```diff
@@ -5,31 +5,39 @@ import { expect, test } from '@playwright/test';
         fullName: 'Top Speaker 1',
         tagLine: 'Visionary Technologist and Thought Leader',
         profileUrl: '/speakers/top-speaker-1-00000000-0000-0000-0000-000000000004/',
+        index: 0,
     },
     {
         fullName: 'Top Speaker 2',
         tagLine: 'Dynamic PR Specialist and Communications Expert',
         profileUrl: '/speakers/top-speaker-2-00000000-0000-0000-0000-000000000008/',
+        index: 1,
     },
     {
         fullName: 'Top Speaker 3',
         tagLine: 'Innovative Tech Visionary and Mentor',
         profileUrl: '/speakers/top-speaker-3-00000000-0000-0000-0000-000000000002/',
+        index: 2,
     },
     {
         fullName: 'Top Speaker 4',
         tagLine: 'Pioneering Innovator in Tech Solutions',
         profileUrl: '/speakers/top-speaker-4-00000000-0000-0000-0000-000000000006/',
+        index: 3,
     },
-].forEach(({ fullName, tagLine, profileUrl }) => {
+].forEach(({ fullName, tagLine, profileUrl, index }) => {
     test(`Should provide link to page of featured speaker ${fullName} on home page`, async ({ page }) => {
         await page.goto('/');
+
         const featuredSpeakersSection = page.getByRole('region', { name: 'Featured Speakers' });
-        const featuredSpeakersMenu = featuredSpeakersSection.getByRole('menu', { name: 'Featured Speakers' });
-        const featuredSpeakersMenuItem = featuredSpeakersMenu.getByRole('menuitem', { name: fullName });
-        await expect(featuredSpeakersMenuItem.getByText(fullName)).toBeVisible();
-        await expect(featuredSpeakersMenuItem.getByText(tagLine)).toBeVisible();
-        await featuredSpeakersMenuItem.getByRole('link', { name: fullName }).click();
+        const featuredSpeakersList = featuredSpeakersSection.getByRole('list', { name: 'Featured Speakers' });
+        const featuredSpeakerItem = featuredSpeakersList.locator(`li[data-testid="speaker-item-${index}"]`);
```

Comment:
Some thoughts on the `locator()`:

- Both, [Testing Library](https://testing-library.com/docs/queries/about#priority) and [Playwright](https://playwright.dev/docs/locators#quick-guide) (they took it over from Testing Library) recommend not to use CSS selectors (or only as last resort). The reason Testing Library exist is that CSS selectors create so many false-positives and false-negatives.
- There is a locator for test IDs (`getByTestId`), but still it's the last option in the list.
- We can still go by role here. From a semantic view, we can expect a list and list items here.

I'll update the test, and then we can close this PR. Great job 🫶🏻